File: javascript01/step05/test06.js
/* 주제: 객체 - 함수를 이용하여 객체를 생성하기

*/

// name:name  왼쪽이 프로퍼티명  오른쪽 변수에 있는 값을 꺼내서 넣는다.
// 직접입력하는값 글자 그대로 입력  Literal

"use strict"

function createScore(name, kor, eng, math) {
  return {
    name: name,
    kor: kor,
    eng: eng,
    math: math,
    sum: kor + eng + math,
    aver: (kor + eng + math) / 3
  }
}

//객체를 리턴하는게 아니라 객체 주소를 리턴한다.

console.log("--------------------------------------------------------------")

var scores2 = [
  createScore("홍길동1", 100, 100, 100),
  createScore("홍길동2", 90, 90, 90),
  createScore("홍길동3", 80, 80, 80),
  createScore("홍길동4", 70, 70, 70),
  createScore("홍길동5", 60, 60, 60)
  //함수를 저장하는게 아니라 함수를 호출한후 리턴값을 놓는다.
]

console.log("홍길동3의 총점은?", scores2[2].sum)

File: java01/src/step05/Test06.java
/* 메서드 : 여러 개의 값을 리턴 하고 싶을 때 IV */
package step05;

public class Test06 {
  static class Student {
    String name;
    int kor;
    int eng;
    int math;
    int sum;
    float aver;
  }
  
  // 1) 사용자(개발자)가 정의한 클래스의 객체로 리턴할 수 있다.
  public static Student getStudent(int StudentNo) {
    Student s = new Student();
    s.name = "홍길동";
    s.kor = 100;
    s.eng = 90;
    s.math = 80;
    s.sum = 100 + 90 + 80;
    s.aver = (100 + 90 + 80 ) /3f;
    
    return s;
  }
  
  public static void main(String[] args) {
   Student s2 = getStudent(100);

   // s2에 저장된 주소의 메모리를 찾아가서 변수에 저장된 값을 찾아서 출력한다.
   // 사용자 정의 객체의 변수 값을 꺼내 출력할 수 있다.
   System.out.println(s2.name);
   System.out.println(s2.kor);
   System.out.println(s2.eng);
   System.out.println(s2.math);
   System.out.println(s2.sum);
   System.out.println(s2.aver);
   
  }
}
